labs-ios-starter/Application/View/UI/Custom Objects/StandardButton.swift

```diff
@@ -0,0 +1,158 @@
+// Copyright © 2020 Shawn James. All rights reserved.
+// StandardButton.swift
+
+import UIKit
+
+/// The button styling used for most buttons
+class StandardButton: UIButton {
+
+    // MARK: - Properties
+        
+    /// used to show loading animation
+    lazy var activityIndicator: UIActivityIndicatorView = {
+        let indicator = UIActivityIndicatorView()
+        indicator.hidesWhenStopped = true
+        indicator.color = .white
+        return indicator
+    }()
+    
+    // MARK: - Initializers
+    
+    /// The initializer called directly by the Developer
+    convenience init(withTitle text: String?, width fixedWidth: CGFloat?, height fixedHeight: CGFloat?) {
+        self.init()
+        
+        if let title = text {
+            setTitle(title, for: .normal)
+        }
+        
+        if let width = fixedWidth {
+            addConstraint(widthAnchor.constraint(equalToConstant: width))
+        }
+        
+        if let height = fixedHeight {
+            addConstraint(heightAnchor.constraint(equalToConstant: height))
+        }
+        
+        translatesAutoresizingMaskIntoConstraints = false
+    }
+    
+    /// Programmatic init
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    /// Storyboard init
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        commonInit()
+    }
+    
+    /// Sets up the button with configuration
+    private func commonInit() {
+        setShadow()
+        setColors()
+        setShape()
+    }
+    
+    // MARK: - Configuration Methods
+    
+    /// Configure the appearance of the button's shadow
+    private func setShadow() {
+        layer.shadowColor = UIColor(red: 0.64, green: 0.71, blue: 0.78, alpha: 1.00).cgColor
+        layer.shadowOffset = CGSize(width: 7, height: 7)
+        layer.shadowRadius = 7
+        layer.shadowOpacity = 0.8
+    }
+    
+    /// Configures the button's color scheme
+    private func setColors() {
+        backgroundColor = .action
+        setTitleColor(.white, for: .normal)
+    }
+    
+    /// Configures the shape of the button
+    private func setShape() {
+        layer.cornerRadius = .cornerRadius
+    }
+    
+    // MARK: - Button Animation Methods
+    
+    /// Button will spring animate to resemble a successful press
+    public func springAnimate() {
```

Comment:
could extend UIButton or even UIView with these animations to avoid repeating their implementation... unless it's different

Reply:
Also, why are these marked public? I believe that's the default unless the class is set otherwise